src/AudioToolbox/AudioFileStream.cs

```diff
@@ -102,19 +104,19 @@ public override string ToString ()
 	}
 
 	public class PacketReceivedEventArgs : EventArgs {
-		public PacketReceivedEventArgs (int numberOfBytes, IntPtr inputData, AudioStreamPacketDescription [] packetDescriptions)
+		public PacketReceivedEventArgs (int numberOfBytes, IntPtr inputData, AudioStreamPacketDescription []? packetDescriptions)
 		{
 			this.Bytes = numberOfBytes;
 			this.InputData = inputData;
 			this.PacketDescriptions = packetDescriptions;
 		}
 		public int Bytes { get; private set; }
 		public IntPtr InputData { get; private set; }
-		public AudioStreamPacketDescription [] PacketDescriptions { get; private set;}
+		public AudioStreamPacketDescription []? PacketDescriptions { get; private set;}
 
 		public override string ToString ()
 		{
-			return String.Format ("Packet (Bytes={0} InputData={1} PacketDescriptions={2}", Bytes, InputData, PacketDescriptions.Length);
+			return String.Format ("Packet (Bytes={0} InputData={1} PacketDescriptions={2}", Bytes, InputData, PacketDescriptions!.Length);
```

Comment:
This might be better with something like this:

```suggestion
			return String.Format ("Packet (Bytes={0} InputData={1} PacketDescriptions={2}", Bytes, InputData, PacketDescriptions?.Length ?? -1);
```

1. It will never throw a NullReferenceException (which ToString shouldn't do)
2. You can deduce from the output that PacketDescriptions is null (if "-1" is printed as the length).